app/scripts/lib/accounts/BalancesController.ts

```diff
@@ -0,0 +1,264 @@
+        unit: string;
+      };
+    };
+  };
+};
+
+/**
+ * Default state of the {@link BalancesController}.
+ */
+export const defaultState: BalancesControllerState = { balances: {} };
+
+/**
+ * Returns the state of the {@link BalancesController}.
+ */
+export type BalancesControllerGetStateAction = ControllerGetStateAction<
+  typeof controllerName,
+  BalancesControllerState
+>;
+
+/**
+ * Updates the balances of all supported accounts.
+ */
+export type BalancesControllerUpdateBalancesAction = {
+  type: `${typeof controllerName}:updateBalances`;
+  handler: BalancesController['updateBalances'];
+};
+
+/**
+ * Event emitted when the state of the {@link BalancesController} changes.
+ */
+export type BalancesControllerStateChange = ControllerStateChangeEvent<
+  typeof controllerName,
+  BalancesControllerState
+>;
+
+/**
+ * Actions exposed by the {@link BalancesController}.
+ */
+export type BalancesControllerActions =
+  | BalancesControllerGetStateAction
+  | BalancesControllerUpdateBalancesAction;
+
+/**
+ * Events emitted by {@link BalancesController}.
+ */
+export type BalancesControllerEvents = BalancesControllerStateChange;
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions = HandleSnapRequest;
+
+/**
+ * Messenger type for the BalancesController.
+ */
+export type BalancesControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  BalancesControllerActions | AllowedActions,
+  BalancesControllerEvents,
+  AllowedActions['type'],
+  never
+>;
+
+/**
+ * {@link BalancesController}'s metadata.
+ *
+ * This allows us to choose if fields of the state should be persisted or not
+ * using the `persist` flag; and if they can be sent to Sentry or not, using
+ * the `anonymous` flag.
+ */
+const balancesControllerMetadata = {
+  balances: {
+    persist: true,
+    anonymous: false,
+  },
+};
+
+const BTC_TESTNET_ASSETS = ['bip122:000000000933ea01ad0ee984209779ba/slip44:0'];
+const BTC_MAINNET_ASSETS = ['bip122:000000000019d6689c085ae165831e93/slip44:0'];
+export const BTC_AVG_BLOCK_TIME = 600000; // 10 minutes in milliseconds
+
+/**
+ * Returns whether an address is on the Bitcoin mainnet.
+ *
+ * This function only checks the prefix of the address to determine if it's on
+ * the mainnet or not. It doesn't validate the address itself, and should only
+ * be used as a temporary solution until this information is included in the
+ * account object.
+ *
+ * @param address - The address to check.
+ * @returns `true` if the address is on the Bitcoin mainnet, `false` otherwise.
+ */
+function isBtcMainnet(address: string): boolean {
+  return address.startsWith('bc1') || address.startsWith('1');
+}
+
+/**
+ * The BalancesController is responsible for fetching and caching account
+ * balances.
+ */
+export class BalancesController extends BaseController<
+  typeof controllerName,
+  BalancesControllerState,
+  BalancesControllerMessenger
+> {
+  #poller: Poller;
+  #isPolling = false;
```

Comment:
This feels unnecessary, since `Poller` already implements this logic too, we could remove it